pkg/controller/defaultdomaincert/default_domain_cert_controller_test.go

```diff
@@ -0,0 +1,875 @@
+		PatchError: errors.New("failed to patch secret"),
+	}
+
+	mockStore := newMockStore()
+	mockStore.setFileContent(testCertPath, []byte(testCertContent))
+	mockStore.setFileContent(testKeyPath, []byte(testKeyContent))
+
+	// Use a fake event recorder to capture events
+	fakeRecorder := &record.FakeRecorder{Events: make(chan string, 10)}
+
+	reconciler := &defaultDomainCertControllerReconciler{
+		client: client,
+		events: fakeRecorder,
+		conf: &config.Config{
+			DefaultDomainCertPath: testCertPath,
+			DefaultDomainKeyPath:  testKeyPath,
+		},
+		store: mockStore,
+	}
+
+	req := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "test-ddc",
+			Namespace: testNamespace,
+		},
+	}
+
+	ctx := context.Background()
+	result, err := reconciler.Reconcile(ctx, req)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "failed to patch secret")
+	assert.Equal(t, ctrl.Result{}, result)
+
+	// Verify that the warning event was recorded
+	select {
+	case event := <-fakeRecorder.Events:
+		assert.Contains(t, event, "Warning")
+		assert.Contains(t, event, "EnsuringCertificateSecretFailed")
+		assert.Contains(t, event, "Failed to ensure Secret for DefaultDomainCertificate")
+		assert.Contains(t, event, "failed to patch secret")
+	case <-time.After(100 * time.Millisecond):
+		t.Fatal("Expected event was not recorded within timeout")
+	}
+}
+
+func TestGetSecret_SuccessfulSecretCreation(t *testing.T) {
+	mockStore := newMockStore()
+	mockStore.setFileContent(testCertPath, []byte(testCertContent))
+	mockStore.setFileContent(testKeyPath, []byte(testKeyContent))
+
+	reconciler := createTestReconciler(nil, mockStore)
+
+	ddc := createTestDefaultDomainCertificate("test-ddc", testNamespace, testSecretName)
+	ddc.UID = "test-uid"
+
+	secret, err := reconciler.getSecret(ddc)
+
+	require.NoError(t, err)
+	require.NotNil(t, secret)
+
+	assert.Equal(t, testSecretName, secret.Name)
+	assert.Equal(t, testNamespace, secret.Namespace)
+	assert.Equal(t, corev1.SecretTypeTLS, secret.Type)
+	assert.Equal(t, []byte(testCertContent), secret.Data["tls.crt"])
+	assert.Equal(t, []byte(testKeyContent), secret.Data["tls.key"])
+	assert.Equal(t, manifests.GetTopLevelLabels(), secret.Labels)
+
+	// Verify owner references
+	assert.Len(t, secret.OwnerReferences, 1)
+	assert.Equal(t, ddc.Name, secret.OwnerReferences[0].Name)
+	assert.Equal(t, ddc.UID, secret.OwnerReferences[0].UID)
+	assert.True(t, *secret.OwnerReferences[0].Controller)
+}
+
+func TestGetSecret_CertificateNotFoundInStore(t *testing.T) {
+	mockStore := newMockStore()
+	// Only set key content, not cert content
+	mockStore.setFileContent(testKeyPath, []byte(testKeyContent))
+
+	reconciler := createTestReconciler(nil, mockStore)
+
+	ddc := createTestDefaultDomainCertificate("test-ddc", testNamespace, testSecretName)
+
+	secret, err := reconciler.getSecret(ddc)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "failed to get certificate from store")
```

Comment:
can we add an assertion specifically for cert not found in store also being contained here?

Reply:
and for other cases where we're concerned with specific errors wile generating the secret